k-distribution/include/builtin/kast.k

```diff
@@ -73,18 +73,16 @@ module KSEQ-SYMBOLIC
   syntax KBott      ::= KConfigVar
   syntax KLabel     ::= #KVariable
 
-  syntax #Pattern ::= "#True" [klabel(#True), symbol]
-  syntax #Pattern ::= "#False" [klabel(#False), symbol]
-  syntax #Pattern ::= K
-                    | "(" #Pattern ")" [bracket]
-                    | "#Not" "(" #Pattern ")" [klabel(#Not), symbol, mlOp]
-                    | "#Ceil" "(" #Pattern ")" [klabel(#Ceil), symbol, mlOp]
-                    > #Pattern "#Equals" #Pattern [klabel(#Equals), symbol, mlOp, format(%i%1%d%n%2%n%i%3%d)]
-                    > #Pattern "#And" #Pattern [klabel(#And), symbol, assoc, comm, unit(#True), mlOp, format(%i%1%d%n%2%n%i%3%d)]
-                    > #Pattern "#Or" #Pattern [klabel(#Or), symbol, assoc, comm, unit(#False), mlOp, format(%i%1%d%n%2%n%i%3%d)]
-                    > #Pattern "#Implies" #Pattern [klabel(#Implies), symbol, mlOp, format(%i%1%d%n%2%n%i%3%d)]
-                    > "#Exists" #KVariable "." #Pattern [klabel(#Exists), symbol, mlOp]
-                    | "#Forall" #KVariable "." #Pattern [klabel(#Forall), symbol, mlOp]
+  syntax KBott ::= "#True" [klabel(#True), symbol]
+  syntax KBott ::= "#False" [klabel(#False), symbol]
+  syntax KBott ::= "#Not" "(" K ")" [klabel(#Not), symbol, mlOp, poly(0, 1)]
+                 | "#Ceil" "(" K ")" [klabel(#Ceil), symbol, mlOp, poly(0, 1)]
```

Comment:
Does this mean that I'll be able to use (all) the ML connectives when writing K definitions?